core/src/main/scala/abel/aggregates/Quantile.scala

```diff
@@ -0,0 +1,35 @@
+package abel.aggregates
+import org.isarnproject.sketches.TDigest
+import spray.json.{JsNumber, JsObject, JsValue}
+
+trait Quantile extends Aggregate[Quantile]
+
+case class Quantile1(value: Long) extends Quantile{
+  override def plus(another: Quantile): Quantile = another match {
+    case other:Quantile1 => QuantileN(TDigest.combine(this.toTDigest, other.toTDigest))
+    case QuantileN(otherTDigest) => QuantileN(TDigest.combine(this.toTDigest,otherTDigest))
+  }
+  def toTDigest = {
+    TDigest.sketch(List(this.value))
+  }
+  override def show: JsValue = JsObject("50" -> JsNumber(value),
```

Comment:
What's the difference between Quantile1 vs QuantileN ? The reason I'm asking is, all the values (different percentiles) of Quantile1 seems to be the same, is that how it's supposed to be?

Reply:
I guess its correct to give percentile as same value for each.

https://en.wikipedia.org/wiki/Percentile
See the nearest-rank method. It takes the `ceil` on the calculated rank. Which corresponds to what @G1GC has done.

The `1` and `N` counterpart is primarily followed when the data-structure is bigger than the value we represent (for ex: hyperloglog has a fixed size representation >= 2 KB mostly but the strings we store in it is way way lesser than that), inspired primarily from `Map1`, `Map2`... from scala.